rusoto/core/src/request.rs

```diff
@@ -260,22 +278,24 @@ impl Stream for HttpClientPayload {
                 } else {
                     Ok(Async::Ready(Some(buffer.split_off(0))))
                 }
-            },
-            SignedRequestPayload::Stream(_, ref mut stream) =>
-                Ok(stream.poll()?)
+            }
+            SignedRequestPayload::Stream(_, ref mut stream) => Ok(stream.poll()?),
         }
     }
 }
 
+/// Default HttpClient type, returned by `HttpClient::new`
+pub type DefaultHttpClient = HttpClient<HttpsConnector<HttpConnector>>;
+
 /// Http client for use with AWS services.
-pub struct HttpClient {
-    inner: HyperClient<HttpsConnector<HttpConnector>, HttpClientPayload>,
-    handle: Handle
+pub struct HttpClient<C> {
```

Comment:
This will be a breaking change. Not sure how we feel about that. I tried to simplify it by putting the `DefaultHttpClient` type in the return statements, but it's still a type change... not sure if there is a non-breaking way to accomplish this.

Reply:
Generally speaking, breaking changes are okay at the moment, and likely will continue to be until we hit 1.0.

In this case however, I believe we can do better by making `HttpsConnector<HttpConnector>` the default type argument for `HttpClient` (i.e. `HttpClient<C = HttpsConnector<HttpConnector>`). This should accomplish two things, firstly it is a backwards-compatible change, and secondly it removes the need for a separate `DefaultHttpClient` type alias. What do you think?

Reply:
Thanks. I’ll try that. I don’t default generic paeans often, but this would be a great case! I’ll get back to you.